Send Images: Matrix -> Signal

nr23730 opened the pull request:
A bit tricky but this finally allows sending images from matrix to signal.

nr23730 committed:
Merge branch 'multiple-attachments'

nr23730 committed:
Add sending of images from matrix to singal

thomas-profitt commented on index.js:
```diff
@@ -23,8 +24,11 @@ class App extends MatrixPuppetBridgeBase {
     this.client.on('message', (ev) => {
       const { source, message } = ev.data;
+      let room = source;
+      if ( message.group != null)
```
This isn't a blocker, but you may want to use brackets for single-statement conditions even though it's syntactically valid without them, because if someone doesn't notice that the brackets are missing, they may add code they think is inside the conditional block, to be surprised when it always executes, or to have it go unnoticed..
```suggestion
      if ( message.group != null) {
```

thomas-profitt commented on index.js:
```diff
@@ -23,8 +24,11 @@ class App extends MatrixPuppetBridgeBase {
       const { source, message } = ev.data;
+      let room = source;
+      if ( message.group != null)
+        room = message.group.id;
```
```suggestion
        room = message.group.id;
    }
```

thomas-profitt commented on index.js:
```diff
@@ -48,19 +56,17 @@ class App extends MatrixPuppetBridgeBase {
-        return this.sendStatusMsg({}, "dont know how to deal with filetype", payload);
+      for ( let i = 0; i < message.attachments.length; i++ ) {
+        let att = message.attachments[i];
+	payload.buffer = new Buffer(att.data);
```
You introduced some tabs here, be sure to convert them back into spaces like `let att = message.attachments[i];` above.
```suggestion
        payload.buffer = new Buffer(att.data);
```

thomas-profitt commented on index.js:
```diff
@@ -71,9 +77,38 @@ class App extends MatrixPuppetBridgeBase {
+  sendImageMessageAsPuppetToThirdPartyRoomWithId(id, data) {
+    return download.getTempfile(data.url, { tagFilename: true }).then(({path}) => {
+      const img = path;
+      let fs = require('fs');
```
Please move requires up top with the others. I'm not sure if it will cause any problems to have this ran each time the function executes (I expect it's OK), but it's convention to keep requires in one place.

thomas-profitt commented on index.js:
```diff
@@ -71,9 +77,38 @@ class App extends MatrixPuppetBridgeBase {
+console.log(id);
+    if ( id.match(/^÷/) ) {
+      console.log("sending to group");
+      return this.client.sendMessageToGroup(this.stringToArrayBuffer(id), text);
```
You shouldn't need to define a function to convert strings to ArrayBuffers if TextEncoder is available in our JavaScript execution environment. Try removing `stringToArrayBuffer` and using it, something like this:
```suggestion
      let idAsArrayBuffer = new TextEncoder().encode(id).buffer;
      return this.client.sendMessageToGroup(idAsArrayBuffer, text);
```

thomas-profitt reviewed (changes requested):
Thanks again for your work!

I hope this review doesn't come off as too negative; I'm really really happy someone's picked up matrix-puppet-signal!

nr23730 commented:
Thanks for looking at my pull request.

I will close this for now and submit a new pull request which will be cleaner and contain your requested changes. During the last days I continued working on the bridge and added some more features but also changed some of the code in this pull request again, so that in my opinion it would make sense to create a new one. 

Currently I'm working on this, but I also look if there is any other good possibility to create a Signal bridge as this one relies on the chrome app which is deprecated for time now and may stop working in near future.
